hibernate-reactive-core/src/main/java/org/hibernate/reactive/loader/ast/internal/ReactiveSingleIdEntityLoaderProvidedQueryImpl.java

```diff
@@ -51,7 +52,7 @@ public CompletionStage<T> load(Object pkValue, LockOptions lockOptions, Boolean
 		query.setParameter( (Parameter<Object>) query.getParameters().iterator().next(), pkValue );
 		query.setHibernateFlushMode( FlushMode.MANUAL );
 
-		return completedFuture( query.uniqueResult() );
+		return (( ReactiveNativeQueryImpl)query).reactiveUnique();
```

Comment:
I haven't checked, but are you sure this method always return a `ReactiveNativeQueryImpl`?

Reply:
Refactored to remove alert

`ReactiveNativeQueryImpl` is the only class/type returned.  All of the other implementations of `reactiveUnique()` end up calling `ReactiveAbstractSelectionQuery.reactiveUnique()`

Reply:
ah... ya, that works. thx